src/app/app.component.ts

```diff
@@ -1,10 +1,84 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { environment } from '../environments/environment';
+
+export interface Item {
+  id: string;
+  name: string;
+  type: string;
+  price: number;
+}
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
-export class AppComponent {
+export class AppComponent implements OnInit {
   title = 'angular-green-grocers';
+  groceries: Item[] = [];
+  cart: { item: Item; quantity: number }[] = [];
+  selectedType: string | null = null;
+
+  constructor(private http: HttpClient) {}
+
+  ngOnInit() {
+    this.fetchGroceries();
+  }
+
+  fetchGroceries() {
+    let apiUrl = environment.apiUrl + 'groceries';
+    if (this.selectedType) {
+      apiUrl += `?type=${this.selectedType}`;
+    }
+    
+    this.http.get<Item[]>(apiUrl).subscribe(
```

Comment:
Remember we need to capture the subscriptions in order to unsubscribe from the, otherwise we risk causing a memory leak